pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -96,8 +96,36 @@ private CalciteSqlParser() {
   private static final Pattern OPTIONS_REGEX_PATTEN =
       Pattern.compile("option\\s*\\(([^\\)]+)\\)", Pattern.CASE_INSENSITIVE);
 
+  /**
+   * Checks for the presence of semicolon in the sql query and modifies the query accordingly
+   *
+   * @param sql sql query
+   * @return sql query without semicolons
+   *
+   */
+  private static String checkForSemicolonInTheQuery(String sql) {
+    // Check if the query has semicolon
+    int semiColonIndex = sql.indexOf(';');
+    if (semiColonIndex > -1) {
+      // Split the input query based on semicolon
+      String[] sqlSplit = sql.split(";");
+
+      // If only semicolons are present in the input query
+      if (sqlSplit.length == 0) {
+        new SqlCompilationException("Caught exception while parsing query: " + sql);
+      } else {
+        // After spliting take the string present at 0th index and then parse the SQL
+        sql = sqlSplit[0];
+      }
```

Comment:
`String.split` is a bit heavy for this. Instead can you do 

```java
int semiColonIndex = sql.indexOf(';');
if (semiColonIndex == 0) {
    throw new SqlCompilationException("SQL query: " + sql + " starts with a semicolon.");
} else if (semiColonIndex > 0 {
    return sql.substring(0, semiColonIndex);
}
return sql;
```

Reply:
Also, rather than just take the first statement, it should maybe throw if there is sql after a semicolon, in case the user thinks that multiple statements can be executed.

Reply:
My suggestion is to change the logic to allow only _terminating_ semicolons (like [beam](https://beam.apache.org/documentation/dsls/sql/calcite/lexical-structure/#terminating-semicolons)) - so only modify the query if there is only whitespace after the semicolon or it is the last character:

```java
int semiColonIndex = sql.indexOf(';');
boolean stripSemiColon = semiColonIndex >= 0;
for (int i = semiColonIndex; i < sql.length() && stripSemiColon; I++) {
  stripSemiColon &= Character.isWhitespace(sql.charAt(i));
}
return stripSemiColon ? sql.substring(0, semiColonIndex) : sql;
```

This way terminating semicolons can be stripped, but rejecting semicolons which can be ambiguous is left to the parser which has the intelligence to discern properly.

Reply:
Updated the logic, really appreciate the feedback